lib/tasks/active_record_views.rake

```diff
@@ -52,12 +52,46 @@ Rake::Task[schema_rake_task].enhance do
     pg_tasks = tasks.send(:class_for_adapter, adapter).new(config)
     pg_tasks.send(:set_psql_env)
 
-    require 'shellwords'
-    system("pg_dump --data-only --table=active_record_views #{Shellwords.escape database} >> #{Shellwords.escape filename}")
-    raise 'active_record_views metadata dump failed' unless $?.success?
+    begin
+      active_record_views_dump = Tempfile.open("active_record_views_dump.sql")
+      require 'shellwords'
+      system("pg_dump --data-only --no-owner --table=active_record_views #{Shellwords.escape database} >> #{Shellwords.escape active_record_views_dump.path}")
+      raise 'active_record_views metadata dump failed' unless $?.success?
 
-    File.open filename, 'a' do |io|
-      io.puts 'UPDATE public.active_record_views SET refreshed_at = NULL WHERE refreshed_at IS NOT NULL;'
+      ActiveRecordViews::RakeTaskUtilities.remove_sql_header_comments(active_record_views_dump.path)
+
+      File.open filename, 'a' do |io|
+        File.foreach(active_record_views_dump.path) do |line|
+          io.puts line
+        end
```

Comment:
I think this can be replaced with a one liner:

```ruby
IO.copy_stream active_record_views_dump, io
```

Reply:
AFAICT that overwrites all of `io` :(

Reply:
Huh. That’d be very much counter to my expectations. It seems to work here when I add the following new expectations (which I probably should have added originally) to the content test:

```ruby
expect(sql).to match(/CREATE TABLE public\.schema_migrations/)
expect(sql).to match(/CREATE VIEW public\.test_view/)
```

Reply:
Not sure where I'm going wrong here. I tried a few different things, a `rewind`, passing the `path`, etc. What test are you putting that in?

Reply:
```patch
diff --git a/lib/tasks/active_record_views.rake b/lib/tasks/active_record_views.rake
index 50a1e02..d1ca92b 100644
--- a/lib/tasks/active_record_views.rake
+++ b/lib/tasks/active_record_views.rake
@@ -60,11 +60,8 @@ Rake::Task[schema_rake_task].enhance do
 
       ActiveRecordViews::RakeTaskUtilities.remove_sql_header_comments(active_record_views_dump.path)
 
-      File.open filename, 'a' do |io|
-        File.foreach(active_record_views_dump.path) do |line|
-          io.puts line
-        end
-
+      File.open filename, 'a+' do |io|
+        IO.copy_stream active_record_views_dump, io
         io.puts 'UPDATE public.active_record_views SET refreshed_at = NULL WHERE refreshed_at IS NOT NULL;'
       end
     ensure
diff --git a/spec/tasks_spec.rb b/spec/tasks_spec.rb
index df18af3..63666e9 100644
--- a/spec/tasks_spec.rb
+++ b/spec/tasks_spec.rb
@@ -77,6 +77,8 @@ describe 'rake tasks' do
 
       expect(File.exist?('spec/internal/db/schema.rb')).to eq false
       sql = File.read('spec/internal/db/structure.sql')
+      expect(sql).to match(/CREATE TABLE public\.schema_migrations/)
+      expect(sql).to match(/CREATE VIEW public\.test_view/)
       expect(sql).to match(/COPY public\.active_record_views.+test_view\tTestView/m)
       expect(sql).to match(/UPDATE public\.active_record_views SET refreshed_at = NULL/)
     end
```

Reply:
Strange, that test fails for me! I'm using Ruby 2.7.3 as I had some issues installing 2.6.7.

Reply:
Huh. Strange indeed. Works on 2.6.7 and fails on 2.7.3. Also failing on 2.7.0 and 3.0.1. I guess the behaviour changed in 2.7.0.

Doing `io.seek(0, IO::SEEK_END)` before `IO.copy_stream` works As does moving the position by `write`ing any data to the IO before calling `copy_stream`. This feels like a Ruby bug to me.

Since the SQL should be plain text, looping over lines will be fine but I think I’d still prefer `copy_stream`, even with an explicit `seek` call to the end as a workaround.

Note: Ignore the mode change from `a` to `a+` in my patch above. I was playing around trying to repo the issue.

Reply:
Pushed a fix!